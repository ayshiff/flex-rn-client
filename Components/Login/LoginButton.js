import { View } from "react-native";
import { Button } from "react-native-elements";
import React from "react";
import styles from "../../views/Login/LoginScreenStyles";
import I18n from "../../i18n/i18n";

const LoginButton = (props: { onPress: () => void }) => {
  const { onPress } = props;
  return (
    <View style={styles.button_container}>
      <Button
        style={styles.button_login}
        fontWeight="bold"
        fontFamily="Raleway"
        borderRadius={15}
        buttonStyle={{
          borderWidth: 1,
          borderColor: "#2E89AD"
        }}
        backgroundColor="#fff"
        color="#2E89AD"
        title={I18n.t("login.title").toUpperCase()}
        onPress={onPress}
        containerStyle={{ marginTop: 20 }}
      />
    </View>
  );
};

export default LoginButton;
