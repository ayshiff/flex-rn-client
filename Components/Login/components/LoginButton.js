import { View } from "react-native";
import { Button } from "react-native-elements";
import React from "react";
import styles from "../LoginScreenStyles";
import I18n from "../../../i18n/i18n";

const LoginButton = (props: { onPress: () => void }) => {
  const { onPress } = props;
  return (
    <View style={styles.button_container}>
      <Button
        style={styles.button_login}
        fontWeight="bold"
        borderRadius={15}
        backgroundColor="#2E89AD"
        color="#fff"
        title={I18n.t("login.title").toUpperCase()}
        onPress={onPress}
      />
    </View>
  );
};

export default LoginButton;
