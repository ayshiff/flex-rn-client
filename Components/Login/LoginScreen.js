// @flow

import React from "react";
import { AsyncStorage, View, Image } from "react-native";

import { Text } from "react-native-elements";
import { omit } from "ramda";
import styles from "./LoginScreenStyles";
import server from "../../config/server";
import config from "../../config/api";
import type { Props, State } from "./LoginScreenType";

import logo from "../../assets/logo.png";

import I18n from "../../i18n/i18n";
import LoginButton from "./components/LoginButton";
import InputLogin from "./components/InputLogin";

import { checkNavigation } from "../../utils/utils";

class LoginScreen extends React.Component<Props, State> {
  static navigationOptions = {
    title: "Flex-Office",
    headerStyle: {
      backgroundColor: "white",
      borderBottomWidth: 8,
      borderBottomColor: "#2E89AD",
      height: 60
    }
  };

  constructor() {
    super();
    this.state = {
      name: "",
      fname: "",
      id: "",
      place: "",
      debug: "",
      debugField: "",
      remoteDay: "",
      historical: [],
      photo: ""
    };
  }

  componentWillMount() {
    fetch(`${server.address}environment`, {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        "x-access-token": config.token
      }
    })
      .then(res => res.json())
      .then(data => {
        const { LOGIN_REGEX, PLACE_REGEX, WIFI_REGEX } = data;
        const environmentVariable = {
          LOGIN_REGEX,
          PLACE_REGEX,
          WIFI_REGEX
        };
        this.setState({ LOGIN_REGEX });
        AsyncStorage.setItem(
          "environment",
          JSON.stringify(environmentVariable)
        );
      });
    checkNavigation(this);
  }

  fetchLoginRegex = async () => {
    const regex = await AsyncStorage.getItem("environment");
    return regex.LOGIN_REGEX;
  };

  logOut = () => {
    AsyncStorage.removeItem("USER");
  };

  /** This function handle the user login */
  logIn() {
    const { navigation } = this.props;
    const { name, fname, id, LOGIN_REGEX, historical } = this.state;

    if (
      name !== "" &&
      fname !== "" &&
      id !== "" &&
      id.match(LOGIN_REGEX) !== null
    ) {
      const payload = {
        name,
        fname,
        id_user: id,
        id_place: "",
        historical
      };

      fetch(`${server.address}login_user`, {
        method: "POST",
        body: JSON.stringify(payload),
        headers: {
          "Content-Type": "application/json",
          "x-access-token": config.token
        }
      })
        .then(res => res.json())
        .then(data => {
          const redirect: boolean = true;
          if (redirect) {
            if (data.user)
              this.setState({
                remoteDay: data.user.remoteDay,
                photo: data.user.photo,
                friend: data.user.friend
              });
            else this.setState({ friend: [] });
            AsyncStorage.setItem(
              "USER",
              JSON.stringify(omit(["debugField"], this.state))
            );
            navigation.navigate("Profile", { photo: data.user.photo });
          }
        });
    } else {
      this.setState({ debugField: I18n.t("login.debug") });
    }
  }

  render() {
    const { debugField } = this.state;
    return (
      <View style={styles.view}>
        <Image source={logo} style={{ height: 120, resizeMode: "contain" }} />
        <View style={styles.view_second}>
          <InputLogin
            onChangeText={text => this.setState({ name: text })}
            onChangeText1={text => this.setState({ fname: text })}
            onChangeText2={text => this.setState({ id: text })}
          />
          <LoginButton onPress={() => this.logIn()} />
          <Text style={styles.debug}>{debugField}</Text>
        </View>
      </View>
    );
  }
}

export default LoginScreen;
