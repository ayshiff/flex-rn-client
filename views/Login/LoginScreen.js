// @flow

import React from "react";
import { AsyncStorage, View, Image } from "react-native";

import { Text } from "react-native-elements";
import { omit } from "ramda";
import LinearGradient from "react-native-linear-gradient";
import styles from "./LoginScreenStyles";
import server from "../../config/server";
import config from "../../config/api";
import type { Props, State } from "./LoginScreenType";

import logo from "../../assets/logo.png";

import I18n from "../../i18n/i18n";
import { checkNavigation } from "../../utils/utils";

/**
 * List of components
 */
// eslint-disable-next-line
import LoginButton from "@components/Login/LoginButton";
// eslint-disable-next-line
import InputLogin from "@components/Login/InputLogin";

const fetchData = function fetchEnvironment() {
  fetch(`${server.address}environment`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
      "x-access-token": config.token
    }
  })
    .then(res => res.json())
    .then(data => {
      const { LOGIN_REGEX, PLACE_REGEX } = data;
      const environmentVariable = {
        LOGIN_REGEX,
        PLACE_REGEX
      };
      this.setState({ LOGIN_REGEX });
      AsyncStorage.setItem("environment", JSON.stringify(environmentVariable));
    });
  checkNavigation(this);
};

class LoginScreen extends React.Component<Props, State> {
  static navigationOptions = {
    header: (
      <View
        style={{
          paddingTop: 20 /* only for IOS to give StatusBar Space */,
          backgroundColor: "white",
          height: 80
        }}
      >
        <View
          style={{
            flex: 1,
            alignItems: "center",
            justifyContent: "center"
          }}
        >
          <Text
            style={{
              color: "black",
              fontWeight: "bold",
              fontSize: 20,
              textAlign: "center",
              fontFamily: "Raleway"
            }}
          >
            Flex-Office
          </Text>
        </View>
        <LinearGradient
          start={{ x: 0, y: 0 }}
          end={{ x: 1, y: 0 }}
          colors={["#58C0D0", "#468BB6", "#3662A0"]}
          style={{ width: "100%", height: 7 }}
        />
      </View>
    )
  };

  constructor() {
    super();
    this.state = {
      name: "",
      fname: "",
      id: "",
      place: "",
      debug: "",
      debugField: "",
      remoteDay: "",
      historical: [],
      photo: ""
    };
  }

  componentWillMount() {
    fetchData.call(this);
  }

  fetchLoginRegex = async () => {
    const regex = await AsyncStorage.getItem("environment");
    return regex.LOGIN_REGEX;
  };

  logOut = () => {
    AsyncStorage.removeItem("USER");
  };

  /** This function handle the user login */
  logIn() {
    const { navigation } = this.props;
    const { name, fname, id, LOGIN_REGEX, historical } = this.state;

    if (
      name !== "" &&
      fname !== "" &&
      id !== "" &&
      id.match(LOGIN_REGEX) !== null
    ) {
      const payload = {
        name,
        fname,
        id_user: id,
        id_place: "",
        historical
      };

      fetch(`${server.address}login_user`, {
        method: "POST",
        body: JSON.stringify(payload),
        headers: {
          "Content-Type": "application/json",
          "x-access-token": config.token
        }
      })
        .then(res => {
          if (res.status === 200) return res.json();
          return false;
        })
        .then(data => {
          if (data !== false) {
            if (data.user)
              this.setState({
                remoteDay: data.user.remoteDay,
                photo: data.user.photo,
                friend: data.user.friend,
                place: data.user.id_place
              });
            else this.setState({ friend: [] });
            AsyncStorage.setItem(
              "USER",
              JSON.stringify(omit(["debugField"], this.state))
            );
            navigation.navigate(
              "Profile",
              data.user ? { photo: data.user.photo } : {}
            );
          } else {
            this.setState({ debugField: I18n.t("login.debug") });
          }
        });
    } else {
      this.setState({ debugField: I18n.t("login.debug") });
    }
  }

  render() {
    const { debugField } = this.state;
    return (
      <View style={styles.view}>
        <Image source={logo} style={{ height: 120, resizeMode: "contain" }} />
        <View style={styles.view_second}>
          <InputLogin
            onChangeText={text => this.setState({ name: text })}
            onChangeText1={text => this.setState({ fname: text })}
            onChangeText2={text => this.setState({ id: text })}
          />
          <LoginButton onPress={() => this.logIn()} />
          <Text style={styles.debug}>{debugField}</Text>
        </View>
      </View>
    );
  }
}

export default LoginScreen;
