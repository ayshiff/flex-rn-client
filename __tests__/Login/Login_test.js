import React from "react";
import "isomorphic-fetch";
import { Image, View } from "react-native";
import enzyme, { shallow } from "enzyme";
import { expect } from "chai";
import ReactSixteenAdapter from "enzyme-adapter-react-16";
import LoginScreen from "../../Components/Login/LoginScreen";
import LoginButton from "../../Components/Login/components/LoginButton";

enzyme.configure({ adapter: new ReactSixteenAdapter() });

const navigation = { navigate: jest.fn(), popToTop: jest.fn() };

it("renders correctly", () => {
  const wrapper = shallow(<LoginScreen navigation={navigation} />);

  wrapper.logIn = jest.fn();
  wrapper.logOut = jest.fn();

  const onPressEvent = jest.fn();

  onPressEvent.mockReturnValue("Link on press invoked");

  // Simulate onPress event on LoginButton component

  wrapper
    .find(LoginButton)
    .first()
    .props()
    .onPress();

  expect(wrapper.find(Image)).to.have.length(1);
  expect(wrapper.find(View)).to.have.length(2);
});
