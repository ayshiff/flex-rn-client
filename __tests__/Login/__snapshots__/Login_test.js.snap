// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      Object {
        "height": 120,
        "resizeMode": "contain",
      }
    }
  />
  <View
    style={
      Object {
        "marginBottom": 30,
        "marginTop": 50,
        "width": 250,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "borderBottomColor": "#bdc6cf",
            "borderBottomWidth": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 10,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Nom"
          style={
            Array [
              Object {
                "color": "#86939e",
                "minHeight": 36,
                "width": 750,
              },
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontFamily": "Raleway",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#bdc6cf",
            "borderBottomWidth": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 10,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="PrÃ©nom"
          style={
            Array [
              Object {
                "color": "#86939e",
                "minHeight": 36,
                "width": 750,
              },
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontFamily": "Raleway",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#bdc6cf",
            "borderBottomWidth": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 10,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="ID"
          style={
            Array [
              Object {
                "color": "#86939e",
                "minHeight": 36,
                "width": 750,
              },
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontFamily": "Raleway",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 50,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            undefined,
            Object {
              "borderRadius": 15,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#fff",
                },
                Object {
                  "borderRadius": 15,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "borderColor": "#2E89AD",
                  "borderWidth": 1,
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#2E89AD",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontFamily": "Raleway",
                    },
                    undefined,
                  ],
                ]
              }
            >
              S'IDENTIFIER
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "color": "#5167A4",
            "margin": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
